src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -149,6 +155,13 @@ def test_reference_is_array():
     assert ref.is_array is True
 
 
+def test_reference_is_character():
+    '''Test that a character reference is marked correctly.
+    '''
+    reference = Reference(DataSymbol("test", CHARACTER_TYPE))
+    assert reference.is_character
```

Comment:
As mentioned before, I would like to see the test extended for a symbol without datatype:
`reference = Reference(Symbol("test")`
and a datatype without intrinsic:
`reference = Reference(DataSymbol("test", UnresolvedType()))`

Reply:
Extended the testing of this functionality in 772d3ff1726e96db078b7d9fbc4e7c0a98e2f3a8.